packages/react-instantsearch-hooks-dom/src/widgets/__tests__/InfiniteHits.test.tsx

```diff
@@ -0,0 +1,214 @@
+            };
+          });
+
+          return createSingleSearchResponse<CustomHit>({
+            hits,
+            page,
+            nbPages: 10,
+            hitsPerPage,
+            index: request.indexName,
+          });
+        })
+      )
+    ),
+});
+
+describe('InfiniteHits', () => {
+  test('renders with default props', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper searchClient={searchClient}>
+        <InfiniteHits />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await waitFor(() => {
+      expect(container.querySelector('.ais-InfiniteHits'))
+        .toMatchInlineSnapshot(`
+        <div
+          class="ais-InfiniteHits"
+        >
+          <ol
+            class="ais-InfiniteHits-list"
+          />
+          <button
+            class="ais-InfiniteHits-loadMore ais-InfiniteHits-loadMore--disabled"
+            disabled=""
+          >
+            Show more results
+          </button>
+        </div>
+      `);
+    });
+  });
+
+  test('renders with a non-default hit shape', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper searchClient={searchClient}>
+        <InfiniteHits<CustomHit>
+          hitComponent={({ hit }) => (
+            <strong>{`${hit.__position} - ${hit.somethingSpecial}`}</strong>
+          )}
+        />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await waitFor(() => {
+      expect(container.querySelectorAll('strong')).toHaveLength(3);
+      expect(container.querySelector('.ais-InfiniteHits'))
+        .toMatchInlineSnapshot(`
+        <div
+          class="ais-InfiniteHits"
+        >
+          <button
+            class="ais-InfiniteHits-loadPrevious ais-InfiniteHits-loadPrevious--disabled"
+            disabled=""
+          >
+            Show previous results
+          </button>
+          <ol
+            class="ais-InfiniteHits-list"
+          >
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <strong>
+                1 - a
+              </strong>
+            </li>
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <strong>
+                2 - b
+              </strong>
+            </li>
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <strong>
+                3 - c
+              </strong>
+            </li>
+          </ol>
+          <button
+            class="ais-InfiniteHits-loadMore"
+          >
+            Show more results
+          </button>
+        </div>
+      `);
+    });
+  });
+
+  test('show more button refines', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper searchClient={searchClient}>
+        <InfiniteHits
+          hitComponent={({ hit: { __position } }) => (
+            <strong>{__position}</strong>
+          )}
+        />
+      </InstantSearchHooksTestWrapper>
+    );
+
```

Comment:
Could we also test with what search parameters `searchClient.search` was called?